index.js

```diff
@@ -703,135 +703,172 @@
 		 *
 		 * @param {unknown} err unknown
 		 *
-		 * @returns {unknown} unknown
+		 * @returns {unknown} A Promise
 		 */
 		function postCallFailure(err) {
-			if (err == 'Sync Error') {
-				this.logger.error(
-					{
-						component: 'semotus',
-						module: 'processCall',
-						activity: 'postCall.syncError',
-						data: {
-							call: remoteCall.name,
-							callTime: logTime(),
-							sequence: remoteCall.sequence
-						}
-					},
-					remoteCall.name
-				);
+			let logString = '';
 
-				packageChanges.call(this, {
+			let packageChangesPayload = {};
+
+			let updateConflictRetry = false;
+
+			if (err == 'Sync Error') {
+				postCallErrorLog(this.logger, 'postCall.syncError', undefined, 'error', remoteCall.name);
+				packageChangesPayload = {
 					type: 'response',
 					sync: false,
-					changes: '',
-					remoteCallId: remoteCallId
-				});
+					changes: ''
+				};
 			} else if (err.message == 'Update Conflict') {
 				// Not this may be caught in the trasport (e.g. Amorphic) and retried)
-				if (callContext.retries++ < 3) {
-					this.logger.warn(
-						{
-							component: 'semotus',
-							module: 'processCall',
-							activity: 'postCall.updateConflict',
-							data: {
-								call: remoteCall.name,
-								callTime: logTime(),
-								sequence: remoteCall.sequence
-							}
-						},
-						remoteCall.name
-					);
 
-					return Q.delay(callContext.retries * 1000).then(retryCall.bind(this));
+				// increment callContext.retries after checking if < 3. Should retry 3 times.
+				if (callContext.retries++ < 3) {
+					postCallErrorLog(this.logger, 'postCall.updateConflict', undefined, 'warn', remoteCall.name);
+					updateConflictRetry = true;
+					// The following assignment is only used for the error handler
+					packageChangesPayload = {
+						type: 'retry'
+					};
 				} else {
-					this.logger.error(
-						{
-							component: 'semotus',
-							module: 'processCall',
-							activity: 'postCall.updateConflict',
-							data: {
-								call: remoteCall.name,
-								callTime: logTime(),
-								sequence: remoteCall.sequence
-							}
-						},
-						remoteCall.name
-					);
-
-					packageChanges.call(this, {
+					postCallErrorLog(this.logger, 'postCall.updateConflict', undefined, 'error', remoteCall.name);
+					packageChangesPayload = {
 						type: 'retry',
-						sync: false,
-						remoteCallId: remoteCallId
-					});
+						sync: false
+					};
 				}
 			} else {
 				if (!(err instanceof Error)) {
-					this.logger.info(
-						{
-							component: 'semotus',
-							module: 'processCall',
-							activity: 'postCall.error',
-							data: {
-								message: JSON.stringify(err),
-								call: remoteCall.name,
-								callTime: logTime(),
-								sequence: remoteCall.sequence
-							}
-						},
-						remoteCall.name
-					);
+					postCallErrorLog(this.logger, 'postCall.error', JSON.stringify(err), 'info', remoteCall.name);
```

Comment:
hmm, not a blocker to merging, but just a thought - if we're going to be defining standard string error levels, what if we put them into some sort of enum? like `const errorLevels = { ERROR: 'error', INFO: 'info' }` and then this might be `postCallErrorLog(this.logger, 'postCall.error', JSON.stringify(err), errorLevels.INFO, remoteCall.name);` or something like that

Reply:
I agree, but I think we need to revisit this more when we rewrite into TS. We should unify ALL logs from semotus